src/runtime/node/timers/internal/immediate.ts

```diff
@@ -1,19 +1,26 @@
 export class Immediate<TArgs extends any[]> implements NodeJS.Immediate {
   _onImmediate: (...args: TArgs) => void;
+  private _timeout: NodeJS.Timeout;
 
   constructor(callback: (...args: TArgs) => void, args: TArgs) {
     this._onImmediate = callback;
-    callback(...args);
+    this._timeout = setTimeout(callback, 0, ...args);
```

Comment:
One additional check  we can also make sure `globalThis.setTimeout` is available and if not, use previous method.